### ISSUE TITLE
Running either the client or the server through the Editor causes connected client to fail.

### ISSUE THREAD

**Issue by robot0x0** [Unity-Technologies/FPSSample#32]
Running the client or the server through the editor (via Project Tools -> Quick Start -> Use Editor as) causes a connected client to throw an exception and fail. The exception is always some form of type mismatch (e.g. "ApplicationException: GAME ASSERT FAILED : Property: has unexpected field type:String Expected:Int"). Please see the full stack trace below.

After doing some digging I've found _what_ is happening but I'm still not sure _why_ it happens. The problem appears to be that the "typeId" for replicated entities gets offset when sending from server to client. This can be seen in NetworkClient.cs:623 and NetworkClient.cs:654 where ClientConnection.ReadSnapshot reads in values for NetworkConfig.schemaTypeIdContext and NetworkConfig.spawnTypeIdContext respectively. As far as I've seen, when the server writes these values they are correct but when the client receives the update the number is offset by 1. As a result, the type mismatch occurs and the client is stuck in the water. For example, when replicating the PlayerState the server sends a "typeId" of 23 (which is correct) but the client receives a "typeId" of 22 (which is actually MovableCube). One more interesting note is that when the server is run in the editor the offset is +1 whereas if the client is run in the editor the offset is -1. Hopefully, this is helpful information. I've reached a point where it's looking like network serialization issue and I'm unfortunately not sure where to start debugging there.

[Full Stack Trace]
```
ApplicationException: GAME ASSERT FAILED : Property: has unexpected field type:String Expected:Int
GameDebug.Assert[T1,T2,T3] (System.Boolean condition, System.String format, T1 arg1, T2 arg2, T3 arg3) (at Assets/Scripts/Core/GameDebug.cs:144)
NetworkReader.ValidateSchema (NetworkSchema+FieldType type, System.Int32 bits, System.Boolean delta, System.Int32 arraySize) (at Assets/Scripts/Networking/NetworkReader.cs:169)
NetworkReader.ReadString (System.Int32 maxLength) (at Assets/Scripts/Networking/NetworkReader.cs:66)
CapturePoint.Deserialize (NetworkReader& reader, IEntityReferenceSerializer refSerializer, System.Int32 tick) (at Assets/Scripts/Game/Systems/GameMode/Components/CapturePoint.cs:41)
ReplicatedEntityCollection.ProcessEntityUpdate (System.Int32 serverTick, System.Int32 id, NetworkReader& reader) (at Assets/Scripts/Game/Modules/ReplicatedEntity/ReplicatedEntityCollection.cs:68)
ReplicatedEntityModuleClient.ProcessEntityUpdate (System.Int32 serverTick, System.Int32 id, NetworkReader& reader) (at Assets/Scripts/Game/Modules/ReplicatedEntity/ReplicatedEntityModuleClient.cs:126)
NetworkClient+ClientConnection.ProcessSnapshot (ISnapshotConsumer consumer) (at Assets/Scripts/Networking/NetworkClient.cs:398)
NetworkClient.ProcessSnapshot (ISnapshotConsumer consumer) (at Assets/Scripts/Networking/NetworkClient.cs:1031)
ClientGameWorld.ProcessSnapshot (System.Int32 serverTick) (at Assets/Scripts/Game/Main/ClientGameLoop.cs:316)
ClientGameLoop.ProcessSnapshot (System.Int32 serverTime) (at Assets/Scripts/Game/Main/ClientGameLoop.cs:1023)
NetworkClient+ClientConnection.ReadPackage[TInputStream] (System.Byte[] packageData, System.Int32 packageSize, NetworkCompression.NetworkCompressionModel compressionModel, INetworkClientCallbacks loop) (at Assets/Scripts/Networking/NetworkClient.cs:440)
NetworkClient.OnData (System.Int32 connectionId, System.Byte[] data, System.Int32 size, INetworkClientCallbacks loop) (at Assets/Scripts/Networking/NetworkClient.cs:302)
NetworkClient.Update (INetworkClientCallbacks loop) (at Assets/Scripts/Networking/NetworkClient.cs:220)
ClientGameLoop.Update () (at Assets/Scripts/Game/Main/ClientGameLoop.cs:602)
Game.Update () (at Assets/Scripts/Game/Main/Game.cs:609)
```

**Comment by robot0x0** [comment]
I found that relaunching Unity, calling UpdateRegistry, rebuilding assets, and rebuilding the game allowed me to run in Editor. During the first time I called UpdateRegistry there was an error so I did it again and it was clean. So, it appears it had to do with a corrupted asset registry which would make some sense that client and server had different "typeId" values.

 I'll close the issue as I can no longer repro but if anyone sees strange type mismatch issues it's likely because of a broken registry which requires a full asset rebuild.
